docs: add section for codegen examples

marionschleifer opened the pull request:
### Description

- This PR adds the structure for adding guides for actions codegen for different frameworks.

- It also adds a first codegen example on implementing custom JWT authentication with codegen using Python / Flask

### Changelog

- [x] `CHANGELOG.md` is updated with user-facing content relevant to this PR.

### Affected components

- [x] Docs

### Affected section

https://deploy-preview-4664--hasura-docs.netlify.app/graphql/manual/actions/codegen/index.html

marionschleifer commented on docs/graphql/manual/actions/codegen/examples/python-flask/index.rst:
```diff
@@ -0,0 +1,387 @@
+
+To test this out, send an HTTP request to your Flask API at ``/signup`` with an email and password. You should get a successful response like this:
+
+.. thumbnail:: ../../../../../../img/graphql/manual/actions/python-flask-signup-request.png
```
@GavinRay97 can we have this as a code snippet? It's nice if users can copy-paste the code to try this out.

marionschleifer commented on docs/graphql/manual/actions/codegen/examples/python-flask/index.rst:
```diff
@@ -0,0 +1,387 @@
+
+Now, if we send a request to ``/login`` using our email and password from ``/signup``, we should successfully get a signed JWT, and decoding the JWT should return the correct information:
+
+.. thumbnail:: ../../../../../../img/graphql/manual/actions/python-request-login.png  
```
Can we have this as a code snippet instead of an image?

marionschleifer commented on docs/graphql/manual/actions/codegen/examples/python-flask/index.rst:
```diff
@@ -0,0 +1,387 @@
+        :alt: Python flask request login
+        :class: no-shadow
+
+.. thumbnail:: ../../../../../../img/graphql/manual/actions/python-flask-jwt-decode.png
```
Can we have this as a code snippet instead of an image?

marionschleifer commented on docs/graphql/manual/actions/codegen/examples/python-flask/index.rst:
```diff
@@ -0,0 +1,387 @@
+Signup Mutation
+^^^^^^^^^^^^^^^
+
+.. thumbnail:: ../../../../../../img/graphql/manual/actions/signup-mutation.png
```
Instead of an image, can we show this in GraphiQL style, as in this example: https://hasura.io/docs/1.0/graphql/manual/queries/simple-object-queries.html#fetch-list-of-objects. Then users can copy paste the signup requests.

marionschleifer commented on docs/graphql/manual/actions/codegen/examples/python-flask/index.rst:
```diff
@@ -0,0 +1,387 @@
+Signup Mutation Error
+^^^^^^^^^^^^^^^^^^^^^
+
+.. thumbnail:: ../../../../../../img/graphql/manual/actions/signup-mutation-duplicate.png
```
Instead of an image, can we show this in GraphiQL style?

marionschleifer commented on docs/graphql/manual/actions/codegen/examples/python-flask/index.rst:
```diff
@@ -0,0 +1,387 @@
+Login Query
+^^^^^^^^^^^
+
+.. thumbnail:: ../../../../../../img/graphql/manual/actions/login-query.png
```
Instead of an image, can we show this in GraphiQL style?

marionschleifer commented on docs/graphql/manual/actions/codegen/examples/python-flask/index.rst:
```diff
@@ -0,0 +1,387 @@
+Login Query Error
+^^^^^^^^^^^^^^^^^
+
+.. thumbnail:: ../../../../../../img/graphql/manual/actions/login-query-bad-password.png
```
Instead of an image, can we show this in GraphiQL style?

GavinRay97 commented:
@marionschleifer 

Removed the section of text, changed all images to either code snippets or GraphiQL :+1: 
https://deploy-preview-4664--hasura-docs.netlify.app/graphql/manual/actions/codegen/examples/python-flask/index.html

Two final small changes were made:

1. I realized the entire app code was never linked and the bits were not necessarily in order, so I put the finished contents of the script in the bottom for easy copy + pasting.

2. I linked to the tutorial from the JWT implementations section, since it's lacking a guide on integrating your own JWT and only mentions how to use Auth-as-a-Service platforms.
 	- https://deploy-preview-4664--hasura-docs.netlify.app/graphql/manual/auth/authentication/jwt.html#auth-jwt-examples

marionschleifer commented:
@GavinRay97 I made the following (and final 😬 ) small changes to the PR:

- Removed unused pictures
- Restructured the page a little
- Added some explanations in between code snippets / examples

Could you please give it a final review making sure everything is still correct and approve if yes?

@praveenweb could you please also give this a review?

GavinRay97 commented:
Hey @marionschleifer, looks good to me :+1: 

One note about the removed images -- they were mostly to provide context for the use of the auto-generated Python dataclasses in the code snippets below. There are lines like `LoginArgs.from_request()` and `User.from_request()` which I thought may be really confusing if you didn't know where those classes came from.

The benefit I can see to removing them is that it's easier to copy + paste the whole tutorial for other languages though.

I think we can merge if everything looks good. Also need to at some point submit the Python + Flask, Ruby + Rails, and Node + Express sample JWT Auth apps to https://github.com/hasura/hasura-actions-examples

marionschleifer commented:
@GavinRay97 regarding the images - can this lost information also be added by code snippets?

GavinRay97 commented:
I don't see why not. Though if they weren't important to the overall understanding of the page and people will get it we can just leave them off :+1:

marionschleifer commented:
@GavinRay97 if you're okay with the changes, can you approve it?

@praveenweb can you please add your review to this too?

praveenweb commented on docs/graphql/manual/actions/codegen/examples/index.rst:
```diff
@@ -0,0 +1,29 @@
+Introduction
+------------
+
+As of now, Hasura provides codegen for a few frameworks (``nodejs-express``,
```
nodejs-express and typescript-zeit are not mentioned in examples though?

marionschleifer replied:
@praveenweb no we have only python-flask for now. @GavinRay97 will add more in the future. I now re-wrote the section a little so that it's less confusing. Please check it out.
